packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -580,7 +582,7 @@ abstract class FlutterCommand extends Command<void> {
         valueHelp: 'x.y.z');
   }
 
-  void usesDartDefineOption() {
+  void usesDartDefineOption({ bool enableDartDefineConfigJsonFileOption = true}) {
```

Comment:
nit make this flag default to false, this way you don't have to edit assemble.dart

Reply:
actually this flag is not used right?

Reply:
> actually this flag is not used right?

Yes, only for `assemble` has no effect

when executing 'flutter assemble', it will not run "packages/flutter_tools/lib/src/runner/flutter_command.dart:1346 "

```
 Future<FlutterCommandResult> verifyThenRunCommand(String? commandPath) async {
    // ……
    //  shouldRunPub is false， it not need pub get and not need buildInfo  
    if (shouldRunPub) {
      // ……
      if (reportNullSafety) {
        await _sendNullSafetyAnalyticsEvents(project);
      }
    }
}

  Future<void> _sendNullSafetyAnalyticsEvents(FlutterProject project) async {
    final BuildInfo buildInfo = await getBuildInfo();
    NullSafetyAnalysisEvent(
      buildInfo.packageConfig,
      buildInfo.nullSafetyMode,
      project.manifest.appName,
      globals.flutterUsage,
    ).send();
  }
```

I personally think that it is not necessary to add it here for the time being, because its construction process is different from others, and it is used less. If the changes will have a great impact, we can decide whether to support it based on the follow-up situation.

This mainly affects the `build`, `run`and other build processes

Reply:
Correct. What I meant was to make `void usesDartDefineOption({ bool enableDartDefineConfigJsonFileOption = false})` and then you don't have to explicitly send a false in `assemble.dart`. My other question is besides assemble.dart where is this function used? I think I don't see any other place